code/script.js

```diff
@@ -1,325 +1,396 @@
-const setSecret = () => {
-  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
-}
+  // Randomly select a person from the characters array and set as the value of the variable called secret
+  const setSecret = () => {
+    secret =
+      charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)];
+  };
 
-// This function to start (and restart) the game
-const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
-}
+  // This function to start (and restart) the game
+  const start = () => {
+    charactersInPlay = CHARACTERS; // Here we're setting charactersInPlay array to be all the characters to start with
+    document.getElementById('winOrLose').style.display = "none";
+    generateBoard(); //call/invoke function generateBoard on row 210
+    setSecret(); //call/invoke function setSecret on row 227
+    console.log("secret:", secret);
+  };
 
-// setting the currentQuestion object when you select something in the dropdown
-const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  // setting the currentQuestion object when you select something in the dropdown
+  const selectQuestion = () => {
+    const category =
+      questions.options[questions.selectedIndex].parentNode.label;
+    //This variable stores what option group (category) the question belongs to.
+    const value = questions.value;
+    /* const value = questions.value Works too */
+    // We also need a variable that stores the actual value of the question we've selected.
 
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+    currentQuestion = {
+      category: category,
+      value: value,
+    };
 
-  currentQuestion = {
-    category: category,
-    // value: value
-  }
-}
+    console.log("question selected:", currentQuestion);
+  };
 
-// This function should be invoked when you click on 'Find Out' button.
-const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  /* questions.onchange = selectQuestion same as 350 */
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  // This function should be invoked when you click on 'Find Out' button.
+  const checkQuestion = () => {
+    const { category, value } = currentQuestion
+  let keep = false
   if (category === 'hair' || category === 'eyes') {
-
+    if (value === secret[category]) {
+      keep = true
+    } else {
+      keep = false
```

Comment:
Since ```let keep = false ``` is declared first in the function, outside the if/else statements, the else statement already has false as default. So this part can be skipped: ```else {
      keep = false}```